### PR TITLE
W1-D2-lambui-view and view group

### PR DESCRIPTION
**Pull request by at-lambui**
Task: #8
anh? xem giúp em bài tập này với . E cảm ơn.

### CODE REVIEW

File: app/src/at-lambui/res/layout/activity_main2.xml [hunk 1]
```diff
@@ -0,0 +1,450 @@
+<?xml version="1.0" encoding="utf-8"?>
+<ScrollView xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    xmlns:android="http://schemas.android.com/apk/res/android">
+
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (at-tainguyen) [inline app/src/at-lambui/res/layout/activity_main2.xml]: WARNING: [UnusedNamespace] Unused namespace declaration xmlns:android; already declared on the root element
 Unused namespace declarations take up space and require processing that is not necessary

Comment (at-lambui) [reply app/src/at-lambui/res/layout/activity_main2.xml]: ok a, e cảm ơn a nhiều nhé.

File: app/src/at-lambui/res/layout/activity_menu.xml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    android:background="@drawable/nancy"
+    xmlns:android="http://schemas.android.com/apk/res/android" >
+    <Button
+        android:id="@+id/btn_Swith"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:textSize="@dimen/font_normal"
+        android:background="@color/colorPrimaryBlack"
+        android:layout_gravity="center"
+        android:text="Bai tap view and view group"/>
```
Comment (at-tainguyen) [inline app/src/at-lambui/res/layout/activity_menu.xml]: WARNING: [HardcodedText] Hardcoded string "Bai tap view and view group", should use `@string` resource
 Hardcoding text attributes directly in layout files is bad for several reasons:

* When creating configuration variations (for example for landscape or portrait)you have to repeat the actual text (and keep it up to date when making changes)

* The application cannot be translated to other languages by just adding new translations for existing string resources.

There are quickfixes to automatically extract this hardcoded string into a resource lookup.

Comment (at-lambui) [reply app/src/at-lambui/res/layout/activity_menu.xml]: e đã fix, cảm ơn a nhé

File: app/src/at-lambui/res/layout/activity_main2.xml [hunk 3]
```diff
@@ -0,0 +1,455 @@
...
+
+        </LinearLayout>
+
+        <LinearLayout
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:id="@+id/llProfile1"
+            android:orientation="horizontal">
+
+            <TextView
```
Comment (at-chauhoang) [inline app/src/at-lambui/res/layout/activity_main2.xml]: Please change ``` TextView  ``` to ``` View ```, It is better than.
Apply for view below.

File: app/src/at-lambui/res/layout/activity_main2.xml [hunk 4]
```diff
@@ -0,0 +1,455 @@
...
+                android:layout_width="8dp"
+                android:layout_height="60dp"
+                android:id="@+id/tvProfile1"
+                android:background="@color/colorWhite"
+                />
```
Comment (at-chauhoang) [inline app/src/at-lambui/res/layout/activity_main2.xml]: Please move to line 294.

File: app/src/at-lambui/res/layout/activity_main2.xml [hunk 5]
```diff
@@ -0,0 +1,455 @@
...
+
+            <de.hdodenhof.circleimageview.CircleImageView
+                xmlns:app="http://schemas.android.com/apk/res-auto"
```
Comment (at-chauhoang) [inline app/src/at-lambui/res/layout/activity_main2.xml]: Please remove it. it already declare above.

File: app/src/at-lambui/res/layout/activity_main2.xml [hunk 6]
```diff
@@ -0,0 +1,455 @@
...
+                android:id="@+id/profile_image2"
+                android:layout_width="@dimen/ic_user_width"
+                android:layout_height="@dimen/ic_user_height"
+                android:layout_marginLeft="30dp"
```
Comment (at-chauhoang) [inline app/src/at-lambui/res/layout/activity_main2.xml]: Please layout_marginLeft =15 dp.
Apply for view below.

File: app/src/at-lambui/res/layout/activity_main2.xml [hunk 7]
```diff
@@ -0,0 +1,455 @@
...
+                android:src="@drawable/hugh"
+                app:civ_border_width="2dp"
+                app:civ_border_color="#FF000000"
+                tools:ignore="DuplicateIds,RtlHardcoded"
+                android:layout_marginStart="30dp" />
```
Comment (at-chauhoang) [inline app/src/at-lambui/res/layout/activity_main2.xml]: Please layout_marginLeft =15 dp.
Apply for view below.

File: app/src/at-lambui/res/layout/activity_main2.xml [hunk 8]
```diff
@@ -0,0 +1,455 @@
...
+
+        <LinearLayout
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:id="@+id/llProfile1"
+            android:orientation="horizontal">
+
+            <TextView
+                android:layout_width="8dp"
+                android:layout_height="60dp"
```
Comment (at-chauhoang) [inline app/src/at-lambui/res/layout/activity_main2.xml]: You should use  layout_height is match_parent,and set hard value at parent view.
Apply for view below.

File: app/src/at-lambui/res/layout/activity_main2.xml [hunk 9]
```diff
@@ -0,0 +1,455 @@
...
+                android:layout_width="@dimen/ic_user_width"
+                android:layout_height="@dimen/ic_user_height"
+                android:layout_marginLeft="30dp"
+                android:src="@drawable/hugh"
+                app:civ_border_width="2dp"
+                app:civ_border_color="#FF000000"
+                tools:ignore="DuplicateIds,RtlHardcoded"
+                android:layout_marginStart="30dp" />
+
+            <LinearLayout
```
Comment (at-chauhoang) [inline app/src/at-lambui/res/layout/activity_main2.xml]: Child view not center, please check.
Apply for view below.

File: app/src/at-lambui/res/layout/activity_main2.xml [hunk 10]
```diff
@@ -0,0 +1,455 @@
...
+                    android:layout_height="wrap_content"
+                    android:textSize="@dimen/font_normal"
+                    android:paddingLeft="@dimen/spacing_large"
+                    android:hint="@string/_1pm_restaurant"
+                    tools:ignore="HardcodedTextRtlSymmetry"
+                    android:paddingStart="@dimen/spacing_large" />
+
+            </LinearLayout>
+
+            <FrameLayout
```
Comment (at-chauhoang) [inline app/src/at-lambui/res/layout/activity_main2.xml]: When small screen, user will can not see fab without scroll to bottom, I want user always see fab on screen.
Please fix it thanks.

File: app/src/at-lambui/res/drawable/custom_button2.xml [hunk 11]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="rectangle">
```
Comment (at-chauhoang) [inline app/src/at-lambui/res/drawable/custom_button2.xml]: Please use android:shape="oval"

File: app/src/at-lambui/res/drawable/custom_button.xml [hunk 12]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="rectangle">
```
Comment (at-chauhoang) [inline app/src/at-lambui/res/drawable/custom_button.xml]: Please use android:shape="oval"

File: app/src/at-lambui/res/drawable/custom_button3.xml [hunk 13]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="rectangle">
```
Comment (at-chauhoang) [inline app/src/at-lambui/res/drawable/custom_button3.xml]: Please use android:shape="oval"

File: app/src/at-lambui/res/values/strings.xml [hunk 14]
```diff
@@ -1,3 +1,29 @@
-<resources>
-    <string name="app_name">Intern18Summer</string>
+<resources xmlns:tools="http://schemas.android.com/tools">
```
Comment (at-chauhoang) [inline app/src/at-lambui/res/values/strings.xml]: Please remove xmlns:tools="http://schemas.android.com/tools".

File: app/src/at-lambui/res/values/strings.xml [hunk 15]
```diff
@@ -1,3 +1,29 @@
...
+    <string name="app_name">Recycleview</string>
+    <string name="menu">menu</string>
+    <string name="todo" tools:ignore="DuplicateDefinition">TODO</string>
```
Comment (at-chauhoang) [inline app/src/at-lambui/res/values/strings.xml]: Remove tools:ignore="DuplicateDefinition"

File: app/src/at-lambui/java/asiantech/internship/summer/Main2Activity.java [hunk 16]
```diff
@@ -0,0 +1,60 @@
...
+import android.view.View;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+public class Main2Activity extends AppCompatActivity implements View.OnClickListener {
+    TextView tvProfile1;
+    TextView tvProfile2;
+    TextView tvProfile3;
+    LinearLayout   llProfile1;
+    LinearLayout   llProfile2;
+    LinearLayout   llProfile3;
```
Comment (at-chauhoang) [inline app/src/at-lambui/java/asiantech/internship/summer/Main2Activity.java]: Please use modifier appropriately.

File: app/src/at-lambui/res/drawable/custom_button.xml (hunk 12 above)
Comment (at-lambui) [reply app/src/at-lambui/res/drawable/custom_button.xml]: a?? chỗ đó  e để rectangle, khi radius = nữa cao, rộng( cao = rộng) là ra hình tròn , nên đổi thành oval rồi custom lại không a???

File: app/src/at-trivo/AndroidManifest.xml [hunk 17]
```diff
@@ -9,7 +9,7 @@
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
         android:theme="@style/AppTheme">
-        <activity android:name=".MainActivity">
+        <activity android:name=".Main2Activity">
```
[the remark this answers is not in the document]
Comment (at-lambui) [reply app/src/at-trivo/AndroidManifest.xml]: Bữa a Tài làm dư main dưới á a, nên nói tạo tên khác tên MainActivity, e đã fix lại tên rồi a nhé, e cảm ơn a nhiều nhé

File: app/src/at-lambui/res/layout/activity_viewandviewgroup.xml [hunk 18]
```diff
@@ -0,0 +1,453 @@
...
+            android:layout_height="wrap_content"
+            android:orientation="horizontal">
+
+            <TextView
+                android:id="@+id/tvProfile2"
+                android:layout_width="8dp"
+                android:layout_height="60dp"
+                android:background="@color/colorWhite" />
+
+            <de.hdodenhof.circleimageview.CircleImageView xmlns:app="http://schemas.android.com/apk/res-auto"
```
Comment (at-chauhoang) [inline app/src/at-lambui/res/layout/activity_viewandviewgroup.xml]: Please remove xmlns:app="http://schemas.android.com/apk/res-auto"
Before you declare it.

File: app/src/at-lambui/res/layout/activity_viewandviewgroup.xml [hunk 19]
```diff
@@ -0,0 +1,453 @@
...
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:hint="@string/_1pm_restaurant"
+                    android:paddingLeft="@dimen/spacing_large"
+                    android:paddingStart="@dimen/spacing_large"
+                    android:textSize="@dimen/font_normal" />
+
+            </LinearLayout>
+
+            <FrameLayout
```
Comment (at-chauhoang) [inline app/src/at-lambui/res/layout/activity_viewandviewgroup.xml]: Fab is always display on screen.
 Current i must scroll to bottom, it display.

File: app/src/at-lambui/res/layout/activity_viewandviewgroup.xml [hunk 20]
```diff
@@ -0,0 +1,453 @@
...
+                android:layout_height="match_parent"
+                android:layout_marginBottom="@dimen/spacing_small"
+                android:paddingLeft="60dp"
+                android:paddingStart="60dp">
+
+                <ImageButton
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:background="@drawable/custom_button_float"
+                    android:contentDescription="@string/todo"
```
Comment (at-chauhoang) [inline app/src/at-lambui/res/layout/activity_viewandviewgroup.xml]: Please remove it.

**Comment by at-chauhoang** [comment]
Please carefully check name of file and variable by convention.

File: app/src/at-lambui/res/layout/activity_viewandviewgroup.xml [hunk 21]
```diff
@@ -0,0 +1,413 @@
...
+
+            </LinearLayout>
+
+        </LinearLayout>
+
+    </ScrollView>
+
+    <android.support.design.widget.FloatingActionButton
+        android:id="@+id/fab"
+        android:clickable="true"
```
Comment (at-dev-support-01) [inline app/src/at-lambui/res/layout/activity_viewandviewgroup.xml]: WARNING: [KeyboardInaccessibleWidget] 'clickable' attribute found, please also add 'focusable'
 A widget that is declared to be clickable but not declared to be focusable is not accessible via the keyboard. Please add the `focusable` attribute as well.

**Pull request merged**